parfive/downloader.py

```diff
@@ -731,6 +749,7 @@ async def _get_ftp(self, session=None, *, url, filepath_partial,
                     return str(filepath)
 
         except Exception as e:
+            remove_file(get_filepath(filepath_partial(None, url), False)[0])
```

Comment:
THis is also wrong but its all I got.

Reply:
For the ftp downloader this isn't actually wrong as the filename can only be computed from the URL (there is no Content-Disposition header to check), but the same logic applies as for the http worker, the `filename` variable is set on line 724 and if the block errors before that point we can't have saved anything to the disk at that path.

Reply:
Done.